tests/scheduler/test_BasisFuncs.py

```diff
@@ -40,6 +40,102 @@ def test_label(self):
+
+        observation = empty_observation()
+        observation["filter"] = "r"
+        observation["note"] = "foo"
+        observation["mjd"] = 59000.0
+
+        visit_gap.add_observation(observation=observation)
+
+        # observation with the wrong note
+        assert visit_gap.check_feasibility(conditions=conditions)
+
+        observation["note"] = "test"
+        visit_gap.add_observation(observation=observation)
+
+        # now observation with the correct note
+        assert not visit_gap.check_feasibility(conditions=conditions)
+
+    def test_visit_gap_with_filter(self):
+
+        visit_gap = basis_functions.VisitGap(note="test", filter_names=["g"])
+
+        conditions = Conditions()
+        conditions.mjd = 59000.0
+
+        # default is feasible
+        assert visit_gap.check_feasibility(conditions=conditions)
+
+        observation = empty_observation()
+        observation["filter"] = "r"
+        observation["note"] = "foo"
+        observation["mjd"] = 59000.0
+
+        visit_gap.add_observation(observation=observation)
+
+        # observation with the wrong note
+        assert visit_gap.check_feasibility(conditions=conditions)
+
+        observation["note"] = "test"
+        visit_gap.add_observation(observation=observation)
+
+        # observation with the wrong filter
+        assert visit_gap.check_feasibility(conditions=conditions)
+
+        observation["filter"] = "g"
+        visit_gap.add_observation(observation=observation)
+
+        # observation with the correct note and filter
+        assert not visit_gap.check_feasibility(conditions=conditions)
+
+    def test_visit_gap_with_multiple_filters(self):
+
+        visit_gap = basis_functions.VisitGap(note="test", filter_names=["g", "i"])
+
+        conditions = Conditions()
+        conditions.mjd = 59000.0
+
+        # default is feasible
+        assert visit_gap.check_feasibility(conditions=conditions)
+
+        observation = empty_observation()
+        observation["filter"] = "r"
+        observation["note"] = "foo"
+        observation["mjd"] = 59000.0
+
+        visit_gap.add_observation(observation=observation)
+
+        # observation with the wrong note
+        assert visit_gap.check_feasibility(conditions=conditions)
+
+        observation["note"] = "test"
+        visit_gap.add_observation(observation=observation)
+
+        # observation with the wrong filter
+        assert visit_gap.check_feasibility(conditions=conditions)
+
+        observation["filter"] = "g"
+        visit_gap.add_observation(observation=observation)
+
+        # observation with the correct note but only one filter
+        assert visit_gap.check_feasibility(conditions=conditions)
+
+        observation["filter"] = "i"
+        visit_gap.add_observation(observation=observation)
+
+        # observation with the correct note and both filters
+        assert not visit_gap.check_feasibility(conditions=conditions)
+
```

Comment:
I would suggest adding a test to check that the visit gap turns back to 'feasible' after the time period has passed -- especially when multiple filters are involved, in case behavior ends up changing (you may want to test in particular, adding enough observations to trigger the "gap" requirement, then adding more observations in both a non-contributing bandpass and a bandpass in filterlist, then checking if those observations at a later time are feasible (in particular if you add an r band and then immediately a g band).